docs/developer-docs/smart-contracts/best-practices/idempotency.mdx

```diff
@@ -56,6 +56,8 @@ Moreover, the ICP/ICRC ledgers use the `created_at_time` parameter to limit the
 
 But even with this improvement used in the ledgers, the time window approach implicitly assumes that the client will be able to get a definite answer to their call within the time window. For example, after the 24 hours expire, the user cannot easily tell if their ledger transfer happened; their only option is to analyze the ledger blocks, which is somewhat tedious, and has to be done carefully to avoid asynchrony issues; see the section on [queryable call results](#queryable-call-results).
 
+Furthermore, just using a time window for deduplication does not guarantee that the memory will stay bounded. In principle, arbitrarily many updates could happen within the time window, although this is bounded in practice by the scaling limits of the ICP. The ICP/ICRC ledgers thus also define a maximum capacity: the number of deduplicated transactions (i.e., deduplication IDs) that they will store in their deduplication store. Further transactions are simply rejected after the capacity is reached, until the window passes and old transactions can be removed from the deduplication store. Yet another extension of the approach is to make the time window simply the guaranteed lower bound, and just store deduplication IDs until the capacity is reached. This way, the clients obtain a hard deduplication guarantee for the time window, and a best-effort attempt to deduplicate transactions even past the window.
```

Comment:
@oggy-dfin Might be missing something but I don't fully understand this, can you explain?
> Yet another extension of the approach is to make the time window simply the guaranteed lower bound, and just store deduplication IDs until the capacity is reached.

What is the "guaranteed lower bound"?

Reply:
You guarantee deduplication up to 24h. Afterwards it might still deduplicate but the transaction might have been evicted from storage if it exceeded the storage capacity. So long as the ledger isn't used much, the deduplication store will not be full and you might end up having deduplication for your transaction for days or weeks.

Reply:
Ah now I get it, thanks!

Would the following wording make this clearer? 
> Yet another extension of the approach is to guarantee deduplication for a given time window, but keep storing deduplication IDs even beyond that window, until the storage capacity is reached.

Reply:
Agree that it's clearer that way, thanks! Adopted with a slight tweak.